File: C++/950. Reveal Cards In Increasing Order.cpp
class Solution {
public:
	vector<int> deckRevealedIncreasing(vector<int>& deck) {
		int n = deck.size();
		vector<int> res(n);
		sort(deck.begin(), deck.end());
		res[n-1] = deck.back();
		for(int i = n - 2; i >= 0; --i){
			res[i] = deck[i];
			rotate(res.begin() + i + 1, res.begin() + n - 1, res.end());
		}
		return res;
	}
};

File: JAVA/950. Reveal Cards In Increasing Order.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Arrays.sort(deck); 
        int n = deck.length;
        int[] result = new int[n];
        Queue<Integer> indexQueue = new LinkedList<>();
        for(int i = 0; i < n; i++) {
            indexQueue.add(i);
        }
        for(int card : deck) {
            result[indexQueue.remove()] = card;
            if(!indexQueue.isEmpty()) {
                indexQueue.add(indexQueue.remove());
            }
        }
        return result;
    }
}
